source/particle/world.cc

```diff
@@ -888,9 +882,9 @@ namespace aspect
               const typename parallel::distributed::Triangulation<dim>::cell_iterator cell (&this->get_triangulation(),
                                                                                             cell_particle->second.first.first,
                                                                                             cell_particle->second.first.second);
-              const std::string cellid = cell->id().to_string();
-              cellid.copy(static_cast<char *>(data),cellid.size());
-              data = static_cast<char *>(data) + cellid.size();
+              const CellId::binary_type cellid = cell->id().template to_binary<dim>();
+              memcpy(data,&(cellid[0]),cellid_size);
```

Comment:
Also, for all practical purposes, `binary_type` is an opaque type, i.e., we don't know whether it's an array. I think it would be more readable to write the second argument simply as `&cellid`.